test/src/kbasesearchengine/test/parse/ObjectParserTest.java

```diff
@@ -0,0 +1,518 @@
+package kbasesearchengine.test.parse;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.CharStreams;
+
+import kbasesearchengine.common.GUID;
+import kbasesearchengine.common.ObjectJsonPath;
+import kbasesearchengine.events.handler.SourceData;
+import kbasesearchengine.parse.IdMapper;
+import kbasesearchengine.parse.ObjectParser;
+import kbasesearchengine.parse.SimpleIdConsumer;
+import kbasesearchengine.parse.SimpleSubObjectConsumer;
+import kbasesearchengine.parse.SubObjectConsumer;
+import kbasesearchengine.system.ObjectTypeParsingRules;
+import kbasesearchengine.system.ObjectTypeParsingRulesFileParser;
+import kbasesearchengine.test.common.TestCommon;
+import us.kbase.common.service.UObject;
+
+public class ObjectParserTest {
+
+    private static Path scratch;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        scratch = Paths.get(TestCommon.getTempDir(), "test_object_parser");
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+
+        final boolean deleteTempFiles = TestCommon.getDeleteTempFiles();
+
+        if (scratch != null && scratch.toFile().exists() && deleteTempFiles) {
+            FileUtils.deleteDirectory(scratch.toFile());
+        }
+    }
+
+    @Test
+    /**
+     * Testing object with 'from-parent' attributes
+     * 
+     * @throws Exception
+     */
+    public void extractParentFragmentGenomeFeatureTest() throws Exception {
+
+        final String jsonResource = "genome01";
+        final String type = "GenomeFeature";
+
+        final String parentJson = extractParentFragment(type, jsonResource);
+        final String expectedParentJson = "{\"domain\":\"B\",\"scientific_name\":\"Shewanella\","
+                + "\"assembly_ref\":\"1/2/1\"}";
+        assertEquals(parentJson, expectedParentJson);
+    }
+
+    @Test
```

Comment:
Minor - I think the convention is to have the @Test and other annotations right above the function signature and not above the javadoc comment.